core/shared/src/main/scala/scalaz/zio/internal/StackBool.scala

```diff
@@ -70,17 +72,17 @@ private[zio] final class StackBool private () {
     if (_size == 0L) b
     else {
       val size  = _size - 1L
-      val index = size & 0XFFL
+      val index = size & 0X3FL
       val entry =
-        if (index == 0L && head.next != null) head.next else head
+        if (index == 63L && head.next != null) head.next else head
 
       ((1L << index) & entry.bits) != 0L
     }
 
   final def popDrop[A](a: A): A = { popOrElse(false); a }
 
   final def toList: List[Boolean] =
-    (0 until _size.toInt).map(getOrElse(_, false)).toList.reverse
```

Comment:
IDK if there are any laws that bind stack and list, but my intuition is that `toList` should yield same result (besides mutation) like `pop`-ing all elements. However non-mutating `toList` is just inefficient and I consider this as "debug" method only.

Reply:
Yes, I was thinking about that too. I had a quick look around and I was seeing different behaviours which made me think that it should just continue the way it behaved before.



`
import java.util.Stack;
import java.util.ArrayList;

public class Main {
    public static void main(String args[]) {
        Stack<Boolean> stack = new Stack<Boolean>();
        stack.push(true);
        stack.push(true);
        stack.push(false);
        stack.push(true);
        stack.push(false);
        ArrayList<Boolean> list = new ArrayList(stack);
        for (int i = 0; i < list.size(); i++) {
          System.out.println(list.get(i));
        }
    }
}

// true
// true
// false
// true
// false
---------------------------------
scala> res27
res29: s.type = Stack(false, true, false, true, true)

scala> res27.toList
res30: List[Boolean] = List(false, true, false, true, true)
---------------------------------
val s = StackBool()

s.push(true)
s.push(true)
s.push(false)
s.push(true)
s.push(false)


println(s.toList)
//List(false, true, false, true, true)
`